### PR TITLE
enh+fix: argparse space action, clean up internal space operations

### PR DESCRIPTION
**Pull request by mgxd**

### CODE REVIEW

File: niworkflows/utils/spaces.py [hunk 1]
```diff
@@ -476,6 +477,14 @@ def get_fs_spaces(self):
...
 
 
+class StoreSpacesAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        spaces = getattr(namespace, self.dest) or []
+        for val in values:
+            spaces.extend(Space.from_string(val))
+        setattr(namespace, self.dest, spaces)
+
+
```
Comment (oesteban) [inline niworkflows/utils/spaces.py:487]: why is this necessary? it seems argparse correctly handles ``Space.from_string`` alone?

File: niworkflows/utils/spaces.py [hunk 2]
```diff
@@ -415,7 +416,7 @@ def __contains__(self, item):
 
     def __repr__(self):
         """Representation of this object."""
-        return 'Imaging spaces: %s' % ', '.join(self._spaces)
+        return 'Imaging spaces: %s' % ', '.join([s.fullname for s in self.spaces])
```
Comment (oesteban) [inline niworkflows/utils/spaces.py]: I agree we need to revise this, but I would go this way (not to drop specs):

```suggestion
        return 'Imaging spaces:%s' % '\n  - '.join([] + [str(s) for s in self.spaces])

File: niworkflows/utils/spaces.py [hunk 3]
```diff
@@ -492,7 +501,6 @@ def _expand_entities(entities):
 
 
     .. testsetup::
-
```
Comment (oesteban) [inline niworkflows/utils/spaces.py]: This line should be here.

File: niworkflows/utils/spaces.py [hunk 4]
```diff
@@ -363,8 +364,8 @@ class SpatialReferences:
     def check_space(space):
         """Build a :class:`Space` object."""
         try:
-            if isinstance(space[0], Space):
-                return space[0]
+            if isinstance(space, Space):
```
Comment (oesteban) [inline niworkflows/utils/spaces.py:367]: I believe ``check_space`` always receives a list of ``Space`` objects.

File: niworkflows/utils/spaces.py (hunk 2 above)
Comment (oesteban) [reply niworkflows/utils/spaces.py]: WDYT?

**Comment by mgxd** [comment]
here's a minimal parsing example I'm using

```python
import argparse
from niworkflows.utils.spaces import Space, SpatialReferences

class StoreSpacesAction(argparse.Action):

    def __call__(self, parser, namespace, values, option_string=None):
        spaces = getattr(namespace, self.dest) or []
        for val in values:
            spaces.extend(Space.from_string(val))
        setattr(namespace, self.dest, spaces)

spaces = SpatialReferences()
parser = argparse.ArgumentParser()
parser.add_argument('--spaces', nargs='+', 
                    #type=Space.from_string,
                    action=StoreSpacesAction,
                    help='output spaces')

pargs = parser.parse_args()
print(pargs.spaces)
if pargs.spaces:
    spaces += pargs.spaces

print(spaces)
```

File: niworkflows/utils/spaces.py (hunk 1 above)
Comment (mgxd) [reply niworkflows/utils/spaces.py:487]: i think it "correctly" handled singleton cases, but would fail when multiple specs where given, i.e. `fsaverage:den-10k:den-164k`

File: niworkflows/utils/spaces.py (hunk 4 above)
Comment (mgxd) [reply niworkflows/utils/spaces.py:367]: shouldn't this take 1 space at a time anyways? so why would we want a list in that case?

File: niworkflows/utils/spaces.py (hunk 2 above)
Comment (mgxd) [reply niworkflows/utils/spaces.py]: sure this is cleaner

File: niworkflows/utils/spaces.py [hunk 5]
```diff
@@ -492,7 +501,6 @@ def _expand_entities(entities):
 
 
     .. testsetup::
```
Comment (mgxd) [inline niworkflows/utils/spaces.py]:
```suggestion
     .. testsetup::
    
```

**Comment by oesteban** [comment]
Can you add your example as a unit test?

**Comment by mgxd** [comment]
@oesteban done

**Comment by mgxd** [comment]
any lingering concerns?
